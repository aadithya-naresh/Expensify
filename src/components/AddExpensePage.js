import React from 'react'

const AddExpensePage = () => (
    <div>
        This is from the add page
    </div>
)

export default AddExpensePage
