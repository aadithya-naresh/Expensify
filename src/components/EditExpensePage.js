import React from 'react'

const EditExpensepage = (props) =>{
    console.log(props)
    return(
    <div>
       Editing {props.match.params.id}
    </div>
    )
}
export default EditExpensepage