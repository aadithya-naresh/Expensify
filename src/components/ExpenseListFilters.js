import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter } from '../actions/filters';

const ExpenseListFilters = (props) => (
  <div>
    <input type="text" value={props.filters.text} onChange={(e) => {
      props.dispatch(setTextFilter(e.target.value))
      console.log(props.state.filters)
    }} />
  </div>
);

const mapStateToProps = (state) => {
  return {
    filters: state.filters,
    state
  };
};

export default connect(mapStateToProps)(ExpenseListFilters);
