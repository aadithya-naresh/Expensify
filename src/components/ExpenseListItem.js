import React from 'react';
import {NavLink} from 'react-router-dom'

export const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <div>
    <h3><NavLink to={`/edit/${id}`}>{description}</NavLink></h3>
    <p>{amount} - {createdAt}</p>
  
  </div>
);

export default ExpenseListItem;