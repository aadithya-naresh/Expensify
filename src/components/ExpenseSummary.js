import React from 'react'
import { connect } from 'react-redux'
import selectExpenses from '../selectors/expenses'
import expensesTotal from '../selectors/expenses-total'
import {Link} from  'react-router-dom'

export const ExpenseSummary = (props) =>{
    const expenseWord = props.expensesCount == 1 ? 'expense' : 'expenses'
   
   return( 
    <div className="page-header">
      <div className="content-container">
      <h1 className="page-header__title">{<p> Viewing <span>{props.expensesCount}</span> {expenseWord} totalling <span>{'â‚¹'+props.expensesTotal.toLocaleString('en-IN')}</span></p>} </h1>
      <div className="page-header__actions">
        <Link className="button" to="/create">Add Expense</Link>
      </div>
      </div>
    </div>
    )
}

const mapStateToProps = (state) => {
    return {
      expensesCount:selectExpenses(state.expenses, state.filters).length,
      expensesTotal: expensesTotal(state.expenses)
    }
}
export default connect(mapStateToProps)(ExpenseSummary);