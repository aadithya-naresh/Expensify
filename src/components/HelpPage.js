import React from 'react'

const HelpPage = () =>(
    <div>
        This is from help page
    </div>
)  

export default HelpPage