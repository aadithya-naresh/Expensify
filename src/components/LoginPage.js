import React from 'react'
import { connect } from 'react-redux'
import {startLogin} from '../actions/auth'

const LoginPage = ({ startLogin }) =>(
    <div>
        <h1>Login Page</h1>
        <button onClick={startLogin}>Login Button</button>
    </div>
)
const mapDispatchToProps = (dispatch) =>({
    startLogin : () => dispatch(startLogin())
})
export default connect(undefined,mapDispatchToProps)(LoginPage)