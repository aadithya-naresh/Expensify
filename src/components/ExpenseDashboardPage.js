import React from 'react'
import ExpenseList from'./ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'
import ExpenseSummary from './ExpenseSummary'

export const ExpenseDashboardPage = () => (
    <div>
    <ExpenseSummary />
    <ExpenseListFilters />
       <ExpenseList />
    </div>
)

export default ExpenseDashboardPage