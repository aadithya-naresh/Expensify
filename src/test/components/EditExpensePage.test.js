import React from 'react'
import {shallow} from 'enzyme'
import { EditExpensePage } from '../../components/EditExpensePage'
import expenses from '../fixtures/expenses'

let editExpense,history,wrapper,removeExpense
beforeEach(() =>{
     editExpense = jest.fn()
     removeExpense = jest.fn()
     history = {push:jest.fn()}
     wrapper = shallow(<EditExpensePage expense={expenses[1]} removeExpense={removeExpense} editExpense={editExpense} history={history}/>)
})

test('should render Edit Expense Page with default values', () =>{
    expect(wrapper).toMatchSnapshot()
})

test('should render edit expense', () =>{
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1])
    expect(history.push).toHaveBeenLastCalledWith('/')
    expect(editExpense).toHaveBeenLastCalledWith(expenses[1].id,expenses[1])
})

test('should render remove expense', () =>{
    wrapper.find('button').simulate('click')
    expect(history.push).toHaveBeenLastCalledWith('/')
    expect(removeExpense).toHaveBeenLastCalledWith(expenses[1].id)
})

