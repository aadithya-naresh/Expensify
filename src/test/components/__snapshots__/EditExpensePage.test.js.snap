// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Edit Expense Page with default values 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 100,
        "createdAt": -345600000,
        "description": "Gum",
        "id": "2",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
